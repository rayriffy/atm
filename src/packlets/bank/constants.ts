export enum Note {
  Five = '5',
  Ten = '10',
  Twenty = '20',
}

export enum ErrorCode {
  InsufficientFunds = 'InsufficientFunds',
  InsufficientBanknotes = 'InsufficientBanknotes',
  CombinationNotFound = 'CombinationNotFound',
}
