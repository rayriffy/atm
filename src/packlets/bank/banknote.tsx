import { Box, Text } from '@chakra-ui/react'
import { memo } from 'react'

import { Note } from './constants'

interface Props {
  type: Note
}

export const Banknote = memo(({ type }: Props) => (
  <Box px={3} py={0.5} border="2px" borderColor="black" rounded="md">
    <Text fontSize="sm" fontWeight="bold">
      {type}
    </Text>
  </Box>
))
