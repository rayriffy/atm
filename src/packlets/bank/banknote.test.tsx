import { render } from '@testing-library/react'
import { describe, it, expect } from 'vitest'

import { Banknote } from './banknote'
import { Note } from './constants'

describe('<Banknote />', () => {
  it.each(Object.values(Note))('should match snapshot (note = %s)', note => {
    const { container } = render(<Banknote type={note} />)
    expect(container).toMatchSnapshot()
  })
})
