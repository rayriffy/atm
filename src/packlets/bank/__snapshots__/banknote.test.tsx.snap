// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Banknote /> > should match snapshot (note = 5) 1`] = `
<div>
  <div
    class="css-prignq"
  >
    <p
      class="chakra-text css-12bse91"
    >
      5
    </p>
  </div>
</div>
`;

exports[`<Banknote /> > should match snapshot (note = 10) 1`] = `
<div>
  <div
    class="css-prignq"
  >
    <p
      class="chakra-text css-12bse91"
    >
      10
    </p>
  </div>
</div>
`;

exports[`<Banknote /> > should match snapshot (note = 20) 1`] = `
<div>
  <div
    class="css-prignq"
  >
    <p
      class="chakra-text css-12bse91"
    >
      20
    </p>
  </div>
</div>
`;
